src/GitHub.TeamFoundation.14/Services/VSGitExt.cs

```diff
@@ -133,10 +133,9 @@ public void JoinTillEmpty()
         async Task<T> GetServiceAsync<T>()
         {
             await JoinableTaskFactory.SwitchToMainThreadAsync();
-            return (T)await asyncServiceProvider.GetServiceAsync(typeof(T));
+            return (T)serviceProvider.GetService(typeof(T));
```

Comment:
This will block the UI thread. Should get `IAsyncServiceProvider` and then use `GetServiceAsync`. That way we can pass in `IServiceProvider` but still use `IAsyncServiceProvider`.

Reply:
On second thoughts, we're waiting until `IGitExt` is available before retrieving it. Using the non-async `GetService` shouldn't be a problem.